x-pack/test/security_solution_api_integration/test_suites/entity_analytics/default_license/risk_engine/risk_engine_privileges.ts

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import expect from '@kbn/expect';
+import type { SecurityService } from '../../../../../../../test/common/services/security/security';
+import { riskEngineRouteHelpersFactory } from '../../utils';
+import { FtrProviderContext } from '../../../../ftr_provider_context';
+
+const USER_PASSWORD = 'changeme';
+const ROLES = [
+  {
+    name: 'security_feature_read',
+    privileges: {
+      kibana: [
+        {
+          feature: {
+            siem: ['read'],
+          },
+          spaces: ['default'],
+        },
+      ],
+    },
+  },
+  {
+    name: 'cluster_manage_index_templates',
+    privileges: {
+      elasticsearch: {
+        cluster: ['manage_index_templates'],
+      },
+    },
+  },
+  {
+    name: 'cluster_manage_transform',
+    privileges: {
+      elasticsearch: {
+        cluster: ['manage_transform'],
+      },
+    },
+  },
+  {
+    name: 'risk_score_index_read',
+    privileges: {
+      elasticsearch: {
+        indices: [
+          {
+            names: ['risk-score.risk-score-*'],
+            privileges: ['read'],
+          },
+        ],
+      },
+    },
+  },
+  {
+    name: 'risk_score_index_write',
+    privileges: {
+      elasticsearch: {
+        indices: [
+          {
+            names: ['risk-score.risk-score-*'],
+            privileges: ['write'],
+          },
+        ],
+      },
+    },
+  },
+];
+
+const ALL_ROLES = ROLES.map((role) => role.name);
+
+const allRolesExcept = (role: string) => ALL_ROLES.filter((r) => r !== role);
+
+const USERNAME_TO_ROLES = {
+  no_cluster_manage_index_templates: allRolesExcept('cluster_manage_index_templates'),
+  no_cluster_manage_transform: allRolesExcept('cluster_manage_transform'),
+  no_risk_score_index_read: allRolesExcept('risk_score_index_read'),
+  no_risk_score_index_write: allRolesExcept('risk_score_index_write'),
+  all: ALL_ROLES,
+};
+
+export default ({ getService }: FtrProviderContext) => {
+  describe('@ess privileges_apis', () => {
+    const supertestWithoutAuth = getService('supertestWithoutAuth');
+    const riskEngineRoutes = riskEngineRouteHelpersFactory(supertestWithoutAuth);
+    const logger = getService('log');
+    let security: SecurityService;
+    try {
+      security = getService('security');
+    } catch (e) {
+      // even though this test doesn't have the @serverless tag I cannot get it to stop running
+      // with serverless config. This is a hack to skip the test if security service is not available
+      logger.info(
+        'Skipping privileges test as security service not available (likely run with serverless config)'
+      );
+      return;
+    }
+
+    const createRole = async ({ name, privileges }: { name: string; privileges: any }) => {
```

Comment:
Much of this functionality already exists in https://github.com/elastic/kibana/blob/main/x-pack/test/common/services/security_solution/roles_users_utils.ts; can you possibly reuse those here?

Reply:
Those methods allow you to create a known role from the ESS and serverless definitions, I would prefer not to pollute the shared roles with a load of niche roles for testing this callout.